osu.Game/Screens/Edit/Compose/Components/ComposeBlueprintContainer.cs

```diff
@@ -91,39 +91,58 @@ protected override void TransferBlueprintFor(HitObject hitObject, DrawableHitObj
             blueprint.DrawableObject = drawableObject;
         }
 
+        private bool nudgeMovementActive;
+
         protected override bool OnKeyDown(KeyDownEvent e)
         {
             if (e.ControlPressed)
             {
                 switch (e.Key)
                 {
                     case Key.Left:
-                        moveSelection(new Vector2(-1, 0));
+                        nudgeSelection(new Vector2(-1, 0));
                         return true;
 
                     case Key.Right:
-                        moveSelection(new Vector2(1, 0));
+                        nudgeSelection(new Vector2(1, 0));
                         return true;
 
                     case Key.Up:
-                        moveSelection(new Vector2(0, -1));
+                        nudgeSelection(new Vector2(0, -1));
                         return true;
 
                     case Key.Down:
-                        moveSelection(new Vector2(0, 1));
+                        nudgeSelection(new Vector2(0, 1));
                         return true;
                 }
             }
 
             return false;
         }
 
+        protected override void OnKeyUp(KeyUpEvent e)
+        {
+            base.OnKeyUp(e);
+
+            if (nudgeMovementActive && !e.ControlPressed)
```

Comment:
This implementation has an unfortunate consequence; namely that if <kbd>Ctrl</kbd> is held after a nudge, you can do other operations on the selected objects and they will be "merged" into the nudge operation.

Example of this:


https://user-images.githubusercontent.com/20418176/216710476-7cd0e014-43dd-4918-993a-83bb0739d6d7.mp4

Sequence of events here:

1. I hold down <kbd>Ctrl</kbd>, do a bunch of nudges
2. I release <kbd>Ctrl</kbd> to conclude the first operation
3. I press <kbd>Ctrl</kbd> again, do a bunch of nudges again
4. This time I keep <kbd>Ctrl</kbd> held, and click on the slider instead - this adds a new control point to the slider
5. When I undo, the control point addition merges with the nudge rather start a separate operation

Not super sure if it's a huge deal, but I figure I better point it out as part of review.

Reply:
Was aware of this, but not sure how much of a big deal it is (nor have an idea in mind on how to better hand it without complicating things a lot).

Reply:
True enough. Maybe fine to go along with this for now until someone complains.